crypto/evp/evp_cnf.c

```diff
@@ -37,13 +42,17 @@ static int alg_module_init(CONF_IMODULE *md, const CONF *cnf)
                 return 0;
             }
             if (m > 0) {
-                EVPerr(EVP_F_ALG_MODULE_INIT, EVP_R_FIPS_MODE_NOT_SUPPORTED);
-                return 0;
+                EVP_set_default_properties(NULL, "fips=yes");
+            } else {
+                /* TODO(3.0): remove the single propery "fips=yes" */
             }
+        } else if (strcmp(oval->name, "default_properties") == 0) {
+            EVP_set_default_properties(NULL, oval->value);
```

Comment:
Should this merge with the `fips=yes`?

I suggest that they should merge or that we lose the fips= version.

The documentation should indicate that the properties replace any previous ones.

Reply:
Should we have functionality to add a property to an already existing global property query spec?  You might note the comment about removing "fips=yes", just a few lines up...

Reply:
And this is a reason for deprecating `fips_mode`.  Its use is disruptive and should be discouraged

Reply:
I'd say we don't but do need to document the replacement.

Reply:
I've added a bit of documentation on this.